app/src/main/java/com/ciandt/dragonfly/example/features/modelselection/ModelSelectionActivity.kt

```diff
@@ -0,0 +1,95 @@
+package com.ciandt.dragonfly.example.features.modelselection
+
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.LinearSnapHelper
+import android.support.v7.widget.RecyclerView
+import android.support.v7.widget.SimpleItemAnimator
+import android.widget.Button
+import com.ciandt.dragonfly.data.Model
+import com.ciandt.dragonfly.example.BaseActivity
+import com.ciandt.dragonfly.example.MainActivity
+import com.ciandt.dragonfly.example.R
+
+
+class ModelSelectionActivity : BaseActivity(), ModelSelectionContract.View {
+
+    private val MODELS_BUNDLE = "MODELS_BUNDLE"
+
+    private lateinit var presenter: ModelSelectionContract.Presenter
+
+    private lateinit var recyclerView: RecyclerView
+    private var models = ArrayList<Model>()
+
+    private lateinit var aboutButton: Button
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_model_selection)
+
+        presenter = ModelSelectionPresenter(this)
+
+        setupList()
+        setupAboutButton()
+
+        if (savedInstanceState != null) {
+            models = savedInstanceState.getParcelableArrayList(MODELS_BUNDLE)
+        } else {
+            presenter.getModelsList()
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle?) {
+        super.onSaveInstanceState(outState)
+        outState?.putParcelableArrayList(MODELS_BUNDLE, models)
+    }
+
+    private fun setupList() {
+        recyclerView = findViewById(R.id.list) as RecyclerView
```

Comment:
I saw this library, but seems like an experiment, not official yet. Since we won't have many views, I found unnecessary.